Makefile

```diff
@@ -24,12 +22,26 @@ fioctl-%:
 		go build $(LDFLAGS) -o bin/$@ main.go
 	@if [ "$@" = "fioctl-windows-amd64" ]; then mv bin/$@ bin/$@.exe; fi
 
+GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
+golangci-lint:
+	@[ -f $(GOLANGCI_LINT) ] || { \
+	set -e ;\
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.51.2 ;\
+	}
+
+.PHONY: lint
+lint: golangci-lint  ## Run linter
+	$(GOLANGCI_LINT) run
+
+.PHONY: lint-fix
+lint-fix: golangci-lint ## Run golangci-lint linter and perform fixes
+	$(GOLANGCI_LINT) run --fix
```

Comment:
```suggestion
# This is the original line, which should stay at the original place; apparently there are important nuances in it.
linter:=$(shell which golangci-lint 2>/dev/null || echo $(HOME)/go/bin/golangci-lint)

has-linter:
	# Another important nuance here is the `test -x` instead of `test -f`.
	# Your check will succeed when a file is not executable; but the command will then fail.
	@test -x $(linter) || (echo 'Please install go linter using "make install-linter"' && exit 1)

install-linter:
	# Here an important nuance is to have an ability to easily upgrade the linter.
	# Your suggested command is a one-time shooter; this makes it eternally usable.
	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(linter)) v1.51.2

# There is little risk to ever have a file named `lint, lint-fix, lint-check`, so I see no necessity to use the `.PHONY` here.
lint-check: has-linter
	$(linter) run

# This may be a lint-fix if you like it more.
I simply prefer shorter names for regularly used commands (`build, check, format, lint`).
lint: has-linter
	$(linter) run --fix
```

A part of this suggestion is "how we did things in the past", and another part is "how I planned to do things" (a part of my unpublished changes for the new build process).
A rationale is present in inline comments where relevant.  These comments should not be a part of the final commit.

Let's talk in Slack if you have questions about some detail.

Reply:
Hi @vkhoroz, 
 
> # This is the original line, which should stay at the original place; apparently there are important nuances in it.
linter:=$(shell which golangci-lint 2>/dev/null || echo $(HOME)/go/bin/golangci-lint)

I see some issues with this one:
- 1 - Not necessary your GOPATH is $(HOME)/go (Because of that the suggestion use gopath)
- 2 - We are adding and using the lint from GOPATH/bin so that NOT necessary will be the version required. In this way, due to the fact that it is a bin used in the project itself (specific version). IMO it should be inside of the project/bin and not change or misconfigure things that are valid for ANY project in the dev env. 

> > # A user may be suprised that `make check` installs software on their laptop.
> # Here we explicitly notify the user what software we need and how to easily install it.
> # The user then gives a conscent to it by explicitly running an install command (or can do it their way).
> has-linter:
> 	# Another important nuance here is the `test -x` instead of `test -f`.
> 	# Your check will succeed when a file is not executable; but the command will then fail.
> 	@test -x $(linter) || (echo 'Please install go linter using "make install-linter"' && exit 1)

That is suck because you need to install the right version for each project everytime that you need to check the stuff. I do not think that is a good DX. 

Because of this, the suggestion is:
- Install the required version into the BIN of the project NOT GOPATH and missed up with your staff
- Install when you run if the specific version not be there already
- We can add an echo "We are installing goling bin into the root project bin to do this check"

> # This may be a lint-fix if you like it more.
> # I simply prefer shorter names for regularly used commands (`build, check, format, lint`).
> lint: has-linter
> 	$(linter) run --fix

We should not run the lint run --fix when we do the check. Why? 
- It will do fix and the dev might not be aware of and do not understand why the check lint passes locally (where the issues are fixed because of this command) and is not passing in the CI. It might take a while until the person figures out to run `git status` and check out the changes. 

Also, I would not recommend trying to fix the issue automatically without the dev knolodge. The dev should use it as a helper when the linter fails to see if is or not possible to solve automatically and be conscious about its usage.

Reply:
Hi @vkhoroz,

I have attempted to implement ALL of your suggestions, with just two points that I hope we can agree weren't addressed exactly as you suggested. Below are the inline comments where I share the rationale behind this.

> linter:=$(shell which golangci-lint 2>/dev/null || echo $(HOME)/go/bin/golangci-lint)

- The version of golangci-lint is specific to each project that we are working on, so I do not agree with, nor like the idea of adding it outside of the project directory where it will be valid for all projects. I hope we can agree to keep it inside the project dir `bin/`.
  
- Additionally, using `$(HOME)/go/bin` does not cover all scenarios. I might not set up my GOPATH in `$(HOME)/go/bin`. For such purposes, I would suggest using the $GOPATH environment variable instead.

> I don't go along with the idea of calling $(linter) run --fix every time that we check the linter

Executing `lint run --fix` during a check seems inadvisable. The developer might not be aware of the fixes and may wonder why the lint check passes locally but fails in CI. It might take some time before the developer realizes the need to run `git status` and inspect the changes.

I am of the opinion that fixing issues should be a conscious decision by the developer, not an automated process during a check. The linter should serve as a helper when issues arise, thereby giving the developer the autonomy to decide whether or not to apply fixes automatically.

Please let me know what do you think now.